modules/beta-private-cluster/cluster.tf

```diff
@@ -428,6 +428,19 @@ resource "google_container_node_pool" "pools" {
       }
     }
 
+    dynamic "linux_node_config" {
+      for_each = contains(
+        keys(
+          merge(
+            local.node_pools_linux_node_configs["all"],
+        local.node_pools_linux_node_configs[each.value["name"]])),
+      "sysctls") ? [1] : []
+
+      content {
+        sysctls = merge(local.node_pools_linux_node_configs["all"], local.node_pools_linux_node_configs[each.value["name"]])["sysctls"]
```

Comment:
for a config like
```tf
all = {
      sysctls = {
        "foo"  = 1
      }
}
pool-01 = {
      sysctls = {
       "bar" = 2
      }
}
```
Wouldn't merge of the second map override the key `sysctls` from the first unless we do 
```
sysctls = merge(lookup(local.node_pools_linux_node_configs["all"],"sysctls",{}),lookup(local.node_pools_linux_node_configs[each.value["name"]],"sysctls",{}))
```

Reply:
Yes, you are right. Are you in favour of keeping this structure and doing this more complex "deep" merge or do you prefer to flatten the structure one level because sysctls is currently the only key in this map?

Reply:
@m10ev I think it makes sense to flatten as only `sysctls` is available.